client/engine/engine.go

```diff
@@ -245,8 +247,55 @@ func (e *Engine) constructObjectiveFromMessage(message protocols.Message) (proto
 			message.SignedStates[0].State(),
 			*e.store.GetAddress(),
 		)
+	case strings.Contains(string(message.ObjectiveId), "Virtual"):
+		initialState := message.SignedStates[0].State()
+
+		if initialState.TurnNum != 0 {
+			return virtualfund.VirtualFundObjective{}, errors.New("cannot construct virtual fund objective without prefund state")
+		}
+		participants := message.SignedStates[0].State().Participants
+		if len(participants) != 3 {
+			return virtualfund.VirtualFundObjective{}, errors.New("a single hop virtual channel must have exactly 3 participants")
+		}
+
+		alice := participants[0]
+		intermediary := participants[1]
+		bob := participants[2]
+		myAddress := *e.store.GetAddress()
+
+		var left *channel.TwoPartyLedger
+		var right *channel.TwoPartyLedger
+		var ok bool
+
+		if alice != myAddress {
+			left, ok = e.store.GetTwoPartyLedger(intermediary, bob)
+			if !ok {
+				return virtualfund.VirtualFundObjective{}, fmt.Errorf("could not find a left ledger channel between %v and %v", alice, intermediary)
+			}
+		}
+		if bob != myAddress {
+			right, ok = e.store.GetTwoPartyLedger(alice, intermediary)
+			if !ok {
+				return virtualfund.VirtualFundObjective{}, fmt.Errorf("could not find a right ledger channel between %v and %v", intermediary, bob)
+			}
+		}
+
+		myRole, err := getMyRole(myAddress, participants)
+		if err != nil {
+			return virtualfund.VirtualFundObjective{}, errors.New("could not determine my role: %w")
+		}
+
+		return virtualfund.New(
+			true, // TODO ensure objective in only approved if the application has given permission somehow
+			message.SignedStates[0].State(),
+			*e.store.GetAddress(),
+			1, // Always a single hop virtual channel
```

Comment:
Is it worth defining this as a constant somewhere?

Reply:
I think this gets resolved by https://github.com/statechannels/go-nitro/pull/284